libnetwork/network.go

```diff
@@ -234,7 +234,7 @@ type network struct {
 	configFrom       string
 	loadBalancerIP   net.IP
 	loadBalancerMode string
-	sync.Mutex
+	mu               sync.Mutex
```

Comment:
nit: there are a lot of methods using the mutex for reading the internal state, could we use `sync.RWMutex`?

Reply:
I recall I had a discussion on that with Cory, and "it depends" (RWMutex can have a higher overhead). That said, if it's mostly reads and only little writes, perhaps it's something to consider.

(I recall I had some old branches on libnetwork where I changed some, but perhaps I never moved them to the moby repo 🤔)

Reply:
https://github.com/moby/moby/pull/43564#discussion_r930446545
https://zephyrtronium.github.io/articles/rwmutex.html

Reply:
That said; we should look at what fields are mutable, and if locking is needed in all places. That will require a bit of work though, as I've encountered "fun" constructs in the past, with things like; https://github.com/moby/moby/blob/868c897ad7e27bcc476a98477549698ae06b563d/libnetwork/network.go#L1173-L1187


Which caused me some head-scratching (`network.createEndpoint()` replaces the `network` with some other network?). I stared at it some times, and it still didn't compute.

Reply:
<img src="https://i.kym-cdn.com/photos/images/newsfeed/001/172/010/6e4.jpg" />

Reply:
Oh wow